gpytorch/lazy/lazy_tensor.py

```diff
@@ -2307,6 +2307,11 @@ def _symeig(self, eigenvectors: bool = False) -> Tuple[Tensor, Optional["LazyTen
     def __matmul__(self, other):
         return self.matmul(other)
 
+    def __rmatmul__(self, other: Tensor) -> Tensor:
+        if other.ndim == 1:
+            return self.transpose(-1, -2).matmul(other)
+        return self.transpose(-1, -2).matmul(other.transpose(-1, -2)).transpose(-1, -2)
+
```

Comment:
I wonder whether it make sense to just introduce a `rmatmul` method on the base lazy tensor that implements this as done here, and then have `__rmatmul__` just call that. That would make it easier to special-case right matmuls for certain lazy tensors by overwriting `rmatmul` and would follow the same API pattern as with most other calls.